pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -180,7 +180,7 @@ func (p *LaunchTemplateProvider) createLaunchTemplate(ctx context.Context, optio
 		},
 		TagSpecifications: []*ec2.TagSpecification{{
 			ResourceType: aws.String(ec2.ResourceTypeLaunchTemplate),
-			Tags:         v1alpha1.MergeTags(v1alpha1.ManagedTagsFor(options.ClusterName), options.Tags),
+			Tags:         v1alpha1.MergeTags(v1alpha1.ManagedTagsFor(options.ClusterName, options.Tags), options.Tags),
```

Comment:
Now that we're passing options.Tags into v1alpha1.ManagedTagsFor, can we collapse this with v1alpha1.MergeTags?

`v1alpha1.TagsFor(ctx, options.Tags) // you can get provisioner and cluster name from ctx`

Reply:
Hi @ellistarn, similar question as previous one. Are you suggesting that MergeTags and ManagedTagsFor be merged into a new TagsFor? If so, seems like we would need to move the validation logic from validateTags into this new TagsFor function? Again, apologies if I'm missing what you are suggesting.

Reply:
@ellistarn when you get a chance, please take a look at the latest push. This logic is now just a single call to MergeTags. The MergeTags now calls ManagedTagsFor to get the base tags, the "Name" tag is added and then the custom tags are merged. This hopefully simplifies the logic, but still allows us to validate for the reserved tags. Thoughts?

Reply:
Done